Round off the standard deviation.

saeta opened the pull request:
Previously, it would print out a huge number of values past the decimal
point. This seems silly to me. This change it to just 2 as a starting point.
(I think ideally in the long term, we should calculate the appropriate
number of significant figures to not be mis-leading. But we can start here.)

Before:

```
name                                                                   time            std                   iterations
-----------------------------------------------------------------------------------------------------------------------
NonBlockingCondition: notify one, no waiters                           49.0 ns         ± 218.03171899854937  1000000
NonBlockingCondition: notify all, no waiters                           48.0 ns         ± 259.1237923013594   1000000
NonBlockingCondition: preWait, cancelWait                              62.0 ns         ± 318.1394131555278   1000000
```

After:

```
name                                                                   time            std           iterations
---------------------------------------------------------------------------------------------------------------
NonBlockingCondition: notify one, no waiters                           55.0 ns         ± 141.28      1000000
NonBlockingCondition: notify all, no waiters                           48.0 ns         ± 285.49      1000000
NonBlockingCondition: preWait, cancelWait                              63.0 ns         ± 262.61      1000000

```

saeta committed:
Round off the standard deviation.

Previously, it would print out a huge number of values past the decimal
point. This seems silly to me. This change it to just 2 as a starting point.
(I think ideally in the long term, we should calculate the appropriate
number of significant figures to not be mis-leading. But we can start here.)

Before:

```
name                                                                   time            std                   iterations
-----------------------------------------------------------------------------------------------------------------------
NonBlockingCondition: notify one, no waiters                           49.0 ns         ± 218.03171899854937  1000000
NonBlockingCondition: notify all, no waiters                           48.0 ns         ± 259.1237923013594   1000000
NonBlockingCondition: preWait, cancelWait                              62.0 ns         ± 318.1394131555278   1000000
```

After:

```
name                                                                   time            std           iterations
---------------------------------------------------------------------------------------------------------------
NonBlockingCondition: notify one, no waiters                           55.0 ns         ± 141.28      1000000
NonBlockingCondition: notify all, no waiters                           48.0 ns         ± 285.49      1000000
NonBlockingCondition: preWait, cancelWait                              63.0 ns         ± 262.61      1000000

```

shabalind commented:
How about we print standard deviation in % instead in XX.XX% format (i.e., 19.25%) ?

This should make it less likely we round off too much.

saeta committed:
Switch to using percentage of median.

This new representation gives a much more effective summary of the performance
and variability of the benchmark. It's also much easier on the eyes too!

Along with this, I switched up the way the stats functions were written (to
instead by computed properties of `Array<Double>`) as that avoided the name
clashes with using similarly named local variables.

Note: while the median is the right measurement, and standard deviation is a useful way
of capturing variation, standard deviation and mean are the statistics that usually
go together.

saeta commented:
New after:

```
name                                                                         time            std         iterations  
-------------------------------------------------------------------------------------------------------------------
AdjacencyList: build a complete graph of size 10                             44533.5 ns      ±  22.26 %  29574       
AdjacencyList: build a complete graph of size 100                            4154168.0 ns    ±   8.74 %  329         
AdjacencyList: build a complete graph of size 1000                           403153815.0 ns  ±   1.20 %  3           
AdjacencyList: DFS on oneEdge graph, counting visitor                        1913199.5 ns    ±  11.25 %  766         
AdjacencyList: BFS on oneEdge graph, counting visitor                        1597807.5 ns    ±   8.78 %  880         
AdjacencyList: Dijkstra on oneEdge graph, counting visitor, constant weight  3328243.5 ns    ±   6.52 %  416        
```

Much better! Just need the right-alignment from https://github.com/google/swift-benchmark/issues/25 ... :-D

saeta committed:
Update tests too.

shabalind commented:
LGTM, looks great! 👍
